### PR TITLE
[StringExtensions] - Minor optimization for ContainsLetter().

### PR DESCRIPTION
**Pull request by ivandrofly**
This is how `String.IsNullOrWhiteSpace` looks like:
```C#
public static bool IsNullOrWhiteSpace(string value)
{
    if (value != null)
    {
        for (int i = 0; i < value.Length; i++)
        {
            if (!char.IsWhiteSpace(value[i]))
            {
                return false;
            }
        }
    }
    return true;
}
```

We were running: `2n`
Optimized to: `n`

### CODE REVIEW

File: libse/StringExtensions.cs [hunk 1]
```diff
@@ -153,13 +153,13 @@ public static string FixExtraSpaces(this string s)
...
-            foreach (var c in s)
+            if (s != null)
             {
-                if (char.IsLetter(c))
-                    return true;
+                foreach (var c in s)
+                {
+                    if (char.IsLetter(c))
+                        return true;
+                }
```
Comment (xylographe) [inline libse/StringExtensions.cs]: Perhaps, to provide real Unicode support (including Asian languages),
```C#
if (s != null)
{
    foreach (var index in StringInfo.ParseCombiningCharacters(s))
    {
        var uc = CharUnicodeInfo.GetUnicodeCategory(s, index);
        if (uc == UnicodeCategory.LowercaseLetter || uc == UnicodeCategory.UppercaseLetter || uc == UnicodeCategory.TitlecaseLetter || uc == UnicodeCategory.ModifierLetter || uc == UnicodeCategory.OtherLetter)
            return true;
    }
    return false;
}
```

**Comment by ivandrofly** [comment]
@niksedk should we use xylographes?

**Comment by niksedk** [comment]
Yes, seems to be more correct :)

**Comment by ivandrofly** [comment]
@niksedk 

Updated :)

**Comment by niksedk** [comment]
I normally prefer to have a method return in the beginning if something is null - rather than to have code nested inside an if-statement, but it's no biggie (and I've not tested performance either)

**Pull request merged**